lib/iris/_shapefiles.py

```diff
@@ -0,0 +1,231 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+
+# Much of this code is originally based off the ASCEND library, developed in
+# the Met Office by Chris Kent, Emilie Vanvyve, David Bentley, Joana Mendes
+# many thanks to them. Converted to iris by Alex Chamberlain-Clay
+
+
+from itertools import product
+import warnings
+
+import dask.array
+import numpy as np
+import shapely
+import shapely.errors
+import shapely.geometry as sgeom
+import shapely.ops
+
+from iris.exceptions import IrisDefaultingWarning
+
+
+def create_shapefile_mask(
+    geometry,
+    cube,
+    minimum_weight=0.0,
+):
+    """Makes a mask for a cube from the shapefile
+
+    Get the mask of the intersection between the
+    given shapely geometry and cube.
+
+    Parameters
+    -----------
+        geometry        : A :class:`shapely.Geometry` object
+
+        cube            : A :class:`iris.cube.Cube`
```

Comment:
I **_think_** this works with a >2D cube, but the context isn't quite clear.
At any rate it should be explained.
TBH though, while `mask_cube` definitely can do this right, i.e. it can broadcast a lower-dimensioned mask array, I'm not sure that this is at all clear from its docs :-(

Reply:
It should work with any cube with x/y coordinates, I've tested with 3d and will test with 4d as well. Added a note to the docstring that may go wrong if other coordinates are the same size as x/y because that might mess up `mask_cube` broadcasting.